### PR TITLE
Consolidate equality logic behind Generic interface

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously the equality operation of verification was implemented
manually for each verifier type. Now the equality verifications all use
the same generic implementation.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/3" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/9" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/10" target="_blank">Graphite</a> 
            * **PR #11** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/11" target="_blank">Graphite</a> 
              * **PR #12** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12" target="_blank">Graphite</a> 
              * **PR #18** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/18" target="_blank">Graphite</a> 
                * **PR #20** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/20" target="_blank">Graphite</a> 
                  * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/21" target="_blank">Graphite</a> 
                    * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/22" target="_blank">Graphite</a> 
                      * **PR #23** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/23" target="_blank">Graphite</a> 
                        * **PR #24** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/24" target="_blank">Graphite</a> 
                          * **PR #25** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/25" target="_blank">Graphite</a> 
                            * **PR #26** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/26" target="_blank">Graphite</a> 
                              * **PR #30** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/30" target="_blank">Graphite</a>  👈
    * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/4" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/30?utm_source=stack-comment).

File: verifier/src/report_body.rs [hunk 1]
```diff
@@ -59,167 +59,129 @@ report_body_field_accessor! {
...
+impl<T> EqualityVerifier<T> {
+    pub fn new(expected: T) -> Self {
+        Self { expected }
     }
 }
 
-impl<T: Accessor<ConfigId>> Verifier<T> for ConfigIdVerifier {
+impl<T, E> Verifier<E> for EqualityVerifier<T>
+where
+    T: Debug + Clone + PartialEq + IntoVerificationError,
```
Comment (nick-mobilecoin) [inline verifier/src/report_body.rs:81]: Note, `PartialEq` here for now until #27 comes in

File: verifier/src/report_body.rs [hunk 2]
```diff
@@ -59,167 +59,129 @@ report_body_field_accessor! {
...
+/// expected [`ConfigSvn`]
+pub type ConfigSvnVerifier = GreaterThanEqualVerifier<ConfigSvn>;
+impl IntoVerificationError for ConfigSvn {
+    fn into_verification_error(expected: Self, actual: Self) -> VerificationError {
+        VerificationError::ConfigSvnTooSmall { expected, actual }
     }
 }
 
-impl<T: Accessor<IsvSvn>> Verifier<T> for IsvSvnVerifier {
+impl<E: Accessor<ConfigSvn>> Verifier<E> for GreaterThanEqualVerifier<ConfigSvn> {
```
Comment (nick-mobilecoin) [inline verifier/src/report_body.rs:132]: manually implementing for each version, until #280 comes in, added a task to #27 to ensure it doesn't get lost.

**Comment by nick-mobilecoin** [comment]
This was done in response to, https://github.com/mobilecoinfoundation/attestation/pull/22#pullrequestreview-1321369983

**Review by jcape — APPROVED** [review]
LGTM, could probably macro-up some tests for all of these as well...

**Review by NotGyro — APPROVED** [review]
Looks good to me

**Pull request merged**